python/ray/serve/api.py

```diff
@@ -365,6 +375,35 @@ def get_handle(
         return handle
 
 
+def _check_http_and_checkpoint_options(
+        client: Client,
+        http_options: Union[dict, HTTPOptions],
+        checkpoint_path: str,
+) -> None:
+    if checkpoint_path and checkpoint_path != client.checkpoint_path:
+        logger.warn(
+            f"The new client checkpoint path '{checkpoint_path}' "
+            f"is different from the existing one '{client.checkpoint_path}'. "
+            "The new checkpoint path is ignored.")
+
+    if http_options:
+        client_http_options = client.http_config
+        new_http_options = http_options if isinstance(
+            http_options, HTTPOptions) else HTTPOptions.parse_obj(http_options)
+        different_fields = []
+        all_http_option_fields = new_http_options.__dict__
+        for field in all_http_option_fields:
+            if getattr(new_http_options, field) != getattr(
```

Comment:
Since we don't know what [middlewares](https://github.com/ray-project/ray/blob/2d58664f98ff62f6c7dbd1290a020f99267af09a/python/ray/serve/config.py#L256) are used, we do a shallow check for each item in the list assuming that the ordering cannot be changed.